qiskit/primitives/containers/bindings_array.py

```diff
@@ -361,26 +364,41 @@ def examine_array(*possible_shapes):
 
     for parameters, val in data.items():
         if len(parameters) != 1:
-            # the last dimension _has_ to be over parameters
+            # Arrays bound to multiple parameters cannot hide their parameter axis.  For example,
+            # values with shape ``(5, 7, 3)`` and parameters ``(a, b, c)`` force the sweep shape to
+            # be ``(5, 7)``.
             examine_array(val.shape[:-1])
         elif val.shape and val.shape[-1] == 1:
-            # this case is a convention, and separated from the previous case for clarity:
-            # if the last axis is 1-d, make an assumption that it is for our 1 parameter
-            examine_array(val.shape[:-1])
+            # Arrays ending in a singleton axis are ambiguous: ``(1,)`` could be a scalar binding or
+            # a length-one sweep, while shapes like ``(N, 1)`` often represent column vectors.  Keep
+            # both interpretations so we can drop the axis when required by other arrays yet still
+            # broadcast with observables such as ``(1, M)``.
+            examine_array(val.shape[:-1], val.shape)
+            if val.shape == (1,):
+                # Explicitly include the scalar interpretation so legacy callers continue to see an
+                # empty sweep shape when nothing else constrains it.
+                examine_array(())
         else:
-            # otherwise, the user has left off the last axis and we'll be nice to them
+            # A trailing axis of size larger than one must correspond to parameters.  For a value
+            # shaped ``(8, 6)`` bound to a single parameter, the leading shape is ``(8,)``.
             examine_array(val.shape)
 
     if only_possible_shapes is None:
         return ()
+    if len(only_possible_shapes) == 0:
+        raise ValueError("Could not find any consistent shape.")
     if len(only_possible_shapes) == 1:
         return next(iter(only_possible_shapes))
-    elif len(only_possible_shapes) == 0:
-        raise ValueError("Could not find any consistent shape.")
-    raise ValueError(
-        "Could not unambiguously determine the intended shape, all shapes in "
-        f"{only_possible_shapes} are consistent with the input; specify shape manually."
-    )
+
+    # Prefer keeping harmless singleton dimensions so column-vector sweeps (``(N, 1)``) can
+    # participate in broadcasting with observable arrays such as ``(1, M)``.  If the ambiguity is
+    # only between scalar interpretations, continue to return the empty shape for backward
+    # compatibility.
+    if () in only_possible_shapes:
```

Comment:
`examine_arrays` performs an intersection, so what's the scenario that allows `()` to co-exist in `only_possible_shapes` together with other things?